### DISCUSSION TITLE
Controller error validation issue with Mui Autocomplete and register

### DISCUSSION THREAD

**Discussion by keithglassford** [react-hook-form/react-hook-form#9644] [Q&A] [answered]
Hi, I have a Material-ui Autocomplete wrapped in a react-hook-form Controller. If I click submit without a value, the formState error is correctly set and the form field is outlined in red. However, if a valid value is then select from the dropdown, the formState isn't revalidated and the error stays after selecting a valid value.

If I type the value, the formState error is reset but not if I set by selection. I have tried all the useForm mode states, setValue etc. but nothing appears to clear the error upon selecting a value.

Following is a code sandbox https://codesandbox.io/s/determined-turing-zg3cwx?file=/src/App.is

An help would be appreciated!

Thanks, Keith

**Comment by Moshyfawn** [comment]
Please, consider posting plane text yourself next time instead of a screenshot ;)

**Reply by keithglassford** [reply]
Apologies Moshyfawn as I copied and pasted from another editor not realizing it was pasted in as photo. Will pay attention in the future.

**Comment by Moshyfawn** [comment, marked as answer]
In [your sandbox](https://codesandbox.io/s/determined-turing-zg3cwx?file=/src/App.is), you're mixing 2 separate API: `Controller` for controlled inputs and `register` for uncontrolled. It's sure to break RHF logic. Additionally, you're using an external React state to control the `Autocomplete` value, which adds to conflicting behaviour of the value

**Reply by Moshyfawn** [reply]
Additionally, I found that the combination of `autoSelect` and `freeSolo` props causes `onChange` value emitted to be either `string` or `object` depending on if the input was changed or blurred. This breaks the MUI input's "is option selected" logic to break like in [this sandbox](https://codesandbox.io/s/cold-morning-8ulr78?file=%2Fsrc%2FApp.js). [This is a sandbox](https://codesandbox.io/s/cold-morning-8ulr78?file=%2Fsrc%2FApp.js) I've updated to use `Controller` API with MUI Autocomplete. You might want to update what value is passed into Autocomplete or if you want the custom user input `string` to be a valid value at all

**Reply by keithglassford** [reply]
Hi Moshyfawn, Thanks so much for your response as your sandbox as it's very helpful;  I understand RHF much better now after studying your example and response!

**Reply by Moshyfawn** [reply]
I'm glad I could help! Don't hesitate to ask any further questions as well, if you have them ;)
